yt/visualization/plot_modifications.py

```diff
@@ -2704,17 +2704,22 @@ def __call__(self, plot):
         max_scale = self.max_frac * xsize
         min_scale = self.min_frac * xsize
 
-        # If no units are set, then identify a best fit distance unit
+        # If no units are set, pick something sensible.
         if self.unit is None:
-            min_scale = plot.ds.get_smallest_appropriate_unit(
-                min_scale, return_quantity=True
-            )
-            max_scale = plot.ds.get_smallest_appropriate_unit(
-                max_scale, return_quantity=True
-            )
-            if self.coeff is None:
-                self.coeff = max_scale.v
-            self.unit = max_scale.units
+            # User has set the axes units and supplied a coefficient.
+            if plot._axes_unit_names is not None and self.coeff is not None:
+                self.unit = plot._axes_unit_names[0]
+            # Nothing provided; identify a best fit distance unit.
```

Comment:
I'm guessing you don't expect `plot._axes_unit_names` and `self.coeff` could be set to None _independently_, and maybe that's perfectly justified with the current state of the code base, but it looks like an implementation detail. I would feel more confident about this `if/else` block if all possibilities were taken into account. I assume that in the conditions I'm describing this should throw an error.